source/features/link-to-changelog-file.tsx

```diff
@@ -15,16 +15,11 @@ interface FileType {
 }
 
 const getCacheKey = (): string => `changelog:${getRepo()!.nameWithOwner}`;
-const changelogNames = new Set(['changelog', 'news', 'changes', 'history', 'release', 'whatsnew']);
 
+const changelogFiles = /^(changelog|news|changes|history|release|whatsnew)(\.(mdx?|mkdn?|mdwn|mdown|markdown|litcoffee|txt|rst))?$/i;
 function findChangelogName(files: string[]): string | false {
-	for (const file of files) {
-		if (changelogNames.has(file.toLowerCase().split('.', 1)[0])) {
-			return file;
-		}
-	}
-
-	return false;
+	const filename = files.find(name => changelogFiles.test(name));
+	return filename ?? false;
```

Comment:
Let’s just change the function’s signature to return `string | undefined` so you can skip this intermediary variable and extra line

Reply:
Done in the new commit.
I also provided an empty string as fallback value `?? ''` in the two function calls.
It should be ok.

Reply:
Nitpick but the fallback value should probably be `false` to keep the logic clear.